packages/plugin-ext/src/hosted/node/scanners/plugin-uri-factory.ts

```diff
@@ -0,0 +1,25 @@
+
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import URI from '@theia/core/lib/common/uri';
+import { PluginPackage } from '../../../common';
+
+export const PluginUriFactory = Symbol('PluginUriFactory');
+
+export interface PluginUriFactory {
```

Comment:
Since you plan on using this in Che, does this makes this API somewhat stable

If so, should you do something like described [here](https://github.com/eclipse-theia/theia/blob/master/doc/api-management.md#stability) so that eventual refactorings don't break it?

Reply:
> Experimental APIs don't require the stability tag

Since it's new API, it can't be stable, since "Experimental APIs can be graduated to stable via the finalization cycle."

Reply:
So no action is the right thing, IMO.

Reply:
Makes sense, thanks.